### PR TITLE
Corrected to force booleans for yes/no questions

### PR DESCRIPTION
**Pull request by ChrisMassengale**
Did some testing, but would have to do much more testing to confirm logic -- though I believe it correct.  This was fairly time consuming.  Add find out about more inherent functions like .empty?  these could be useful.

@mechamanny - any input would be welcome.  Thanks

### CODE REVIEW

File: ruby/vampires.rb [hunk 1]
```diff
@@ -14,6 +14,19 @@ def validboolean(mydata)
   end
 end
 
+def convertboolean(mydata)
```
Comment (mechamanny) [inline ruby/vampires.rb]: ruby convention is to use `snake_case` when naming a variable with two or more words.

File: ruby/vampires.rb [hunk 2]
```diff
@@ -14,6 +14,19 @@ def validboolean(mydata)
...
+  if mydata == String then
```
Comment (mechamanny) [inline ruby/vampires.rb]: have you tested to see if `"a string" == String` returns true? It does not for me.

File: ruby/vampires.rb [hunk 3]
```diff
@@ -14,6 +14,19 @@ def validboolean(mydata)
...
+  
+  case mydata
+    when 'y', 'yes', 'true', true
+      true
+    else
+      false
+  end #case
+end 
+
 def VampStatus (name, age_provided, year_born, likes_garlic, requires_health_ins, allergy)
```
Comment (mechamanny) [inline ruby/vampires.rb]: method names are `lower_case` and `snake_case`

File: ruby/vampires.rb [hunk 4]
```diff
@@ -59,92 +72,55 @@ def print_vamp_inputs (employee, age, bday, garlic_eater, aca_user, allergy)
...
-  #Main Program --  should this be a Method of its own?  
-  puts "How many employees are to be processed today?"
-  employee_count = gets.chomp.to_i
-
-  #for loop to process multiple employees
-  for i in 1 .. employee_count
-
-    puts "Please Enter your Name:"
+#for loop to process multiple employees
+for i in 1 .. employee_count
```
Comment (mechamanny) [inline ruby/vampires.rb]: check out the `times` method here

File: ruby/vampires.rb [hunk 5]
```diff
@@ -59,92 +72,55 @@ def print_vamp_inputs (employee, age, bday, garlic_eater, aca_user, allergy)
...
+  begin
```
Comment (mechamanny) [inline ruby/vampires.rb]: `begin` and the `for` loop are doing the same job, consolidate!

File: ruby/vampires.rb [hunk 6]
```diff
@@ -59,92 +72,55 @@ def print_vamp_inputs (employee, age, bday, garlic_eater, aca_user, allergy)
...
+    puts "Please Enter Employee Name:"
     employee_name = gets.chomp
+  end until !(employee_name.empty?)
```
Comment (mechamanny) [inline ruby/vampires.rb]: You could write a check like this `until employee_name` since a string (or any data type except false) are 'truthy' in Ruby.

Comment (mechamanny) [inline ruby/vampires.rb]: As long as employee name was nil beforehand.

File: ruby/vampires_withtest.rb [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+  end 
+
+  return vamp_status
+end
+
+def print_vamp_inputs (employee, age, bday, garlic_eater, aca_user, allergy)
+    puts "\nVamp Inputs"
+    puts "==========="
+    p employee
```
Comment (mechamanny) [inline ruby/vampires_withtest.rb]: Try to keep `p` statements for debugging and `puts` for displaying to the console.

File: ruby/vampires_withtest.rb [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+  testsubjects.push ["TU Fang", 20, 1996, "yes", "yes", "none", "Definitely a vampire."]
+  testsubjects.push ["Howdy Dootie", 20, 1996, "yes", "yes", "none", "Probably NOT a vampire."]
+  testsubjects.push ["Chucky", 22, 1996, "yes", "no", "none", "Probably a vampire."]
+  testsubjects.push ["Chucky", 22, 1996, "no", "yes", "none", "Probably a vampire."]
+  testsubjects.push ["Freddie", 22, 1996, "no", "no", "none", "Almost certainly is a vampire."]
+
+  # puts testsubjects
+  # puts testsubjects.length
+  counter = 0
+  testsubjects.each do |(x,y,z,a,b,c,d)|
```
Comment (mechamanny) [inline ruby/vampires_withtest.rb]: These variable names mean nothing, be more specific.

File: ruby/vampires_withtest.rb [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    p bday
+    p garlic_eater
+    p aca_user
+    p allergy
+end
+
+def testvampstatus
+  # Array [name, age, birthdate, likes garlic?, health insurance?, allergy, expected result]
+  # Note currently allergy is not a list of allergies it only matters if it is sunshine or not.
+  testsubjects = Array.new
```
Comment (mechamanny) [inline ruby/vampires_withtest.rb]: This data would be better represented as a hash, something like :

`test_subject1 = { name: "chris", age: 20 .. }` etc. Then you could iterate over it more succinctly.

File: ruby/vampires_withtest.rb [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+end #def tempvampstatus
+
+#================
+#Troubleshooting -- change perform_test to true.
+perform_test = false
+
+# p perform_test
+
+if perform_test != true then
```
Comment (mechamanny) [inline ruby/vampires_withtest.rb]: I dont see where this would be true, what is the reason for this check?

File: ruby/vampires.rb (hunk 2 above)
Comment (ChrisMassengale) [inline ruby/vampires.rb]: I thought I had, but in irb, it doesn't return true as you pointed out.

File: ruby/vampires_withtest.rb (hunk 10 above)
Comment (ChrisMassengale) [inline ruby/vampires_withtest.rb]: When I was troubleshooting with test data  I just manually changed perform_test from true to false on line 95.   I was trying the most expedient way I could find to test the results.  I did not remove the logic after running the tests.  

I responded to two of your comments, but have read the others -- and understood your recommendations.  I will do my best to apply them to future work.   Thanks.

**Pull request merged**